overlord/snapstate/snapstate_install_test.go

```diff
@@ -4215,3 +4215,13 @@ func (s *snapmgrTestSuite) TestInstallPrerequisiteWithSameDeviceContext(c *C) {
 		{macaroon: s.user.StoreMacaroon, name: "snap-content-slot", target: filepath.Join(dirs.SnapBlobDir, "snap-content-slot_11.snap")},
 	})
 }
+
+func (s *snapmgrTestSuite) TestInstallDeduplicatesSnapNames(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	installed, ts, err := snapstate.InstallMany(s.state, []string{"some-snap", "some-base", "some-snap", "some-base"}, s.user.ID)
+	c.Assert(err, IsNil)
+	c.Check(installed, testutil.DeepUnsortedMatches, []string{"some-snap", "some-base"})
```

Comment:
you can probably use DeepEquals as the order is preserved
```suggestion
	c.Check(installed, DeepEquals, []string{"some-snap", "some-base"})
```